src/styles/main.scss

```
@@ -1,3 +1,67 @@
-body {
+$box-size: 100px;
+$margin-top: 80px;
+$gap: 1px;
+$calendar-size: calc(($box-size + $gap) * 7);
+$week-days: ('mon':0, 'tue':1, 'wed':2, 'thu':3, 'fri':4, 'sat':5, 'sun':6);
+$font-size30: 30px;
+
+* {
   margin: 0;
+  box-sizing: border-box;
+}
+
+body {
+  height: 100vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  width: $calendar-size - $gap;
+  &__day {
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    height: $box-size;
+    width: $box-size;
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.2s;
+    &:hover {
+      cursor: pointer;
+      background-color: #ffbfcb;
+      transform: translate(0, -20px);
+      transition-duration: 0.5s;
+    }
+
+    @for $index from 1 through  31 {
+      &:nth-child(#{$index}) {
+        &::before {
+          content: "#{$index}";
+          font-family: Arial, Helvetica, sans-serif;
+          font-size: $font-size30;
+        }
+      }
+    }
+  }
+}
+
+.calendar--start-day {
+  @each $day, $index in $week-days {
+    &-#{$day} > :first-child {
+      margin-left: $index * ($box-size + $gap);
+    }
+  }
+}
+
+.calendar--month-length {
+  @for $i from 28 through 31 {
+    &-#{$i} > :nth-child(n + #{$i + 1}) {
+      display: none;
+    }
+  }
 }
```

it could be nested in calendar
```suggestion

.calendar {
//all styles for calendar

  &--start-day {
    @each $day, $index in $week-days {
      &-#{$day} > :first-child {
        margin-left: $index * ($box-size + $gap);
      }
    }
  }
  
  &--month-length {
    @for $i from 28 through 31 {
      &-#{$i} > :nth-child(n + #{$i + 1}) {
        display: none;
      }
    }
  }
 
}
```